app/src/org/odk/collect/android/jr/extensions/IntentCallout.java

```diff
@@ -126,70 +129,70 @@ public boolean processResponse(Intent intent, ODKView currentView, FormInstance
                     key = "";
                 }
 
-                //Figure out where it's going
-                TreeReference ref = responses.get(key);
+                for (TreeReference ref : responses.get(key)) {
```

Comment:
If you don't mind, it would be sweet to lift all the logic in this loop into a helper method. It pretty burly at the moment.